contracts/modules/TransferManager/BlacklistTransferManager.sol

```diff
@@ -0,0 +1,268 @@
+        emit ModifyBlacklistType(_startTime, _endTime, _name, _repeatPeriodTime);
+    }
+
+    /**
+    * @notice Used to delete the blacklist type
+    * @param _name name of the blacklist type
+    */
+    function deleteBlacklistType(bytes32 _name) public withPerm(ADMIN){
+        require(blacklists[_name].endTime != 0, "Blacklist type doesn’t exist");
+        require(blacklistToInvestor[_name].length == 0, "Investors are associated with the blacklist");
+        // delete blacklist type 
+        delete(blacklists[_name]);
+        emit DeleteBlacklistType(_name);
+    }
+
+    /**
+    * @notice Used to assign the blacklist type to the investor
+    * @param _investor address of the investor
+    * @param _blacklistName name of the blacklist
+    */
+    function addInvestorToBlacklist(address _investor, bytes32 _blacklistName) public withPerm(ADMIN){
+        require(blacklists[_blacklistName].endTime != 0, "Blacklist type doesn't exist");
+        require(_investor != address(0), "Invalid investor address");
+        uint256 investorIndex = investorToBlacklist[_investor].length;
+        // Add blacklist index to the investor 
+        investorToIndex[_investor][_blacklistName] = investorIndex;
+        uint256 blacklistIndex = blacklistToInvestor[_blacklistName].length;
+        // Add investor index to the blacklist
+        blacklistToIndex[_blacklistName][_investor] = blacklistIndex;
+        investorToBlacklist[_investor].push(_blacklistName);
+        blacklistToInvestor[_blacklistName].push(_investor);
+        emit AddInvestorToBlacklist(_investor, _blacklistName);
+    }
+
+    /**
+    * @notice Used to delete the investor from the all associated blacklists
+    * @param _investor address of the investor
+    */
+    function deleteInvestorFromAllBlacklist(address _investor) public withPerm(ADMIN){
+        require(_investor != address(0), "Invalid investor address");
+        require(investorToBlacklist[_investor].length != 0, "Investor is not associated to any blacklist type");
+        for(uint256 i = 0; i < investorToBlacklist[_investor].length; i++){
+            deleteInvestorFromBlacklist(_investor, investorToBlacklist[_investor][i]);
+        }
+    }
+
+    /**
+    * @notice Used to delete the investor from the blacklist
+    * @param _investor address of the investor
+    * @param _blacklistName name of the blacklist
+    */
+    function deleteInvestorFromBlacklist(address _investor,bytes32 _blacklistName) public withPerm(ADMIN){
+        require(_investor != address(0), "Invalid investor address");
+        require(_blacklistName != bytes32(0),"Invalid blacklist name");
+        require(investorToBlacklist[_investor][investorToIndex[_investor][_blacklistName]] != bytes32(0), "Investor is not associated to blacklist type");
+        // delete the investor from the blacklist type
+        delete(blacklistToInvestor[_blacklistName][blacklistToIndex[_blacklistName][_investor]]);
```

Comment:
It would be better to completely remove the entry from the lists (`investorToBlacklist`, `blacklistToInvestor`) as we loop over `investorToBlacklist` in `verifyTransfer`. Generally you can do this by switching the last member of the list to the entry being removed, and update it's index if needed.